pydantic/main.py

```diff
@@ -748,6 +748,13 @@ def __setattr__(self, name: str, value: Any) -> None:
                 'input': value,
             }
             raise pydantic_core.ValidationError.from_exception_data(self.__class__.__name__, [error])
+        elif name in self.model_fields and self.model_fields[name].frozen:
```

Comment:
let's not look up `self.model_fields` twice.

Reply:
```suggestion
        elif getattr(self.model_fields.get(name), "frozen", False)
```

Would this be ok? Or else I need to make sure `name` is part of `self.model_fields`